### PR TITLE
Filter out Stripe client_id

### PR DESCRIPTION
**Pull request by dacook**
#### What? Why?

- Follow-up from https://github.com/openfoodfoundation/openfoodnetwork/pull/12018

#### What should we test?
Stripe specs pass

#### Dependencies
Ideally we'd deal with this before the next Stripe version bump.

### CODE REVIEW

File: spec/support/vcr_setup.rb [hunk 1]
```diff
@@ -7,8 +7,10 @@
...
-  config.filter_sensitive_data('<HIDDEN-STRIPE_SECRET_TEST_API_KEY>') { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
-  config.filter_sensitive_data('<HIDDEN-STRIPE_CUSTOMER>') { ENV.fetch('STRIPE_CUSTOMER', nil) }
-  config.filter_sensitive_data('<HIDDEN-STRIPE_ACCOUNT>') { ENV.fetch('STRIPE_ACCOUNT', nil) }
   config.ignore_hosts('localhost', '127.0.0.1', '0.0.0.0', 'api.knapsackpro.com')
+
+  # Filter sensitive environment variables
+  ['STRIPE_SECRET_TEST_API_KEY', 'STRIPE_CUSTOMER', 'STRIPE_ACCOUNT'].each do |env_var|
+    config.filter_sensitive_data("<HIDDEN-#{env_var}>") { ENV.fetch(env_var, nil) }
+  end
 end
```
Comment (filipefurtad0) [inline spec/support/vcr_setup.rb:16]: Extra smile for that commit message :-D

Comment (filipefurtad0) [reply spec/support/vcr_setup.rb:16]: Oh, but it looks like the line is too long... And rubocop complains. Other than that, looks great to me!

Comment (dacook) [reply spec/support/vcr_setup.rb:21]: 🤦  Oops, fixed now.

**Review by dacook — COMMENTED** [review]
As an aside, I'm actually not sure why we need to filter out "bogus_customer" and "bogus_account", as those are stored in `.env.test`.

File: spec/support/vcr_setup.rb [hunk 2]
```diff
@@ -7,8 +7,15 @@
...
+
+  # Filter sensitive environment variables
+  [
+    'STRIPE_SECRET_TEST_API_KEY',
+    'STRIPE_CUSTOMER',
+    'STRIPE_ACCOUNT',
+    'STRIPE_CLIENT_ID',
+  ].each do |env_var|
+    config.filter_sensitive_data("<HIDDEN-#{env_var}>") { ENV.fetch(env_var, nil) }
+  end
```
Comment (mkllnk) [inline spec/support/vcr_setup.rb:20]: Nice! I often like to simplify it a little bit:
```suggestion
  %w[
    STRIPE_SECRET_TEST_API_KEY
    STRIPE_CUSTOMER
    STRIPE_ACCOUNT
    STRIPE_CLIENT_ID
  ].each do |env_var|
    config.filter_sensitive_data("<HIDDEN-#{env_var}>") { ENV.fetch(env_var, nil) }
  end
```

Slightly cleaner and easier to add a new item.

**Review by mkllnk — APPROVED** [review]
Excellent!

**Review by filipefurtad0 — APPROVED** [review]
:raised_hands:

**Pull request merged**